Content.Server/Atmos/Reactions/PlasmaFireReaction.cs

```diff
@@ -9,77 +10,79 @@ namespace Content.Server.Atmos.Reactions;
+        IGasMixtureHolder? holder,
+        AtmosphereSystem atmosphereSystem,
+        float heatScale)
     {
-        var energyReleased = 0f;
-        var oldHeatCapacity = atmosphereSystem.GetHeatCapacity(mixture, true);
         var temperature = mixture.Temperature;
-        var location = holder as TileAtmosphere;
         mixture.ReactionResults[(byte)GasReaction.Fire] = 0;
 
-        // More plasma released at higher temperatures.
-        var temperatureScale = 0f;
-
-        if (temperature > Atmospherics.PlasmaUpperTemperature)
-            temperatureScale = 1f;
-        else
-        {
-            temperatureScale = (temperature - Atmospherics.PlasmaMinimumBurnTemperature) /
-                               (Atmospherics.PlasmaUpperTemperature - Atmospherics.PlasmaMinimumBurnTemperature);
-        }
-
-        if (temperatureScale > 0)
+        // React faster and with less oxygen per plasma at higher temperatures, up to a maximum.
+        var temperatureReactionScale = ProportionOfRange(temperature,
+            Atmospherics.PlasmaMinimumBurnTemperature,
+            Atmospherics.PlasmaUpperTemperature);
+        if (temperatureReactionScale > 0)
         {
-            var oxygenBurnRate = Atmospherics.OxygenBurnRateBase - temperatureScale;
-            var plasmaBurnRate = 0f;
-
             var initialOxygenMoles = mixture.GetMoles(Gas.Oxygen);
             var initialPlasmaMoles = mixture.GetMoles(Gas.Plasma);
 
-            // Supersaturation makes tritium.
-            var oxyRatio = initialOxygenMoles / initialPlasmaMoles;
-            // Efficiency of reaction decreases from 1% Plasma to 3% plasma:
-            var supersaturation = Math.Clamp((oxyRatio - Atmospherics.SuperSaturationEnds) /
-                                             (Atmospherics.SuperSaturationThreshold -
-                                              Atmospherics.SuperSaturationEnds), 0.0f, 1.0f);
-
-            if (initialOxygenMoles > initialPlasmaMoles * Atmospherics.PlasmaOxygenFullburn)
-                plasmaBurnRate = initialPlasmaMoles * temperatureScale / Atmospherics.PlasmaBurnRateDelta;
-            else
-                plasmaBurnRate = temperatureScale * (initialOxygenMoles / Atmospherics.PlasmaOxygenFullburn) / Atmospherics.PlasmaBurnRateDelta;
+            // The maximum rate based on the relative proportions of reactants, expressed as moles of plasma burned.
+            var maximumReactionRate =
+                Math.Min(initialPlasmaMoles, initialOxygenMoles / Atmospherics.PlasmaOxygenFullburn);
+            var reactionRate = maximumReactionRate * temperatureReactionScale / Atmospherics.PlasmaBurnRateDelta;
 
-            if (plasmaBurnRate > Atmospherics.MinimumHeatCapacity)
+            // TODO I have no fucking idea why the rate is compared against a heat capacity here.
+            // ... but it has the effect of not doing little finicky adjustments, I guess.
+            if (reactionRate > Atmospherics.MinimumHeatCapacity)
```

Comment:
Should `Atmospherics.MinimumHeatCapacity` be replaced by something else here?
Not to say the value should change, but it really doesn't make sense that this is comparing a number of moles to a specific heat.